assets/js/event-handler.js

```diff
@@ -1,8 +1,11 @@
 "use-strict";
 
+// Ensure dataLayer is defined Globally before using it
+window.dataLayer = window.dataLayer || [];
```

Comment:
dataLayer has not been used in the changed code anywhere. Why are we re-initializing it? This may affect other code which may be doing similar checks.
If you want to avoid errors which may arise due to pushing into an undefined dataLayer object, you may add a conditionalto check if dataLayer is defined before you push into it

Reply:
This code was modified to fix the error mentioned in the issue https://github.com/virtual-labs/ph3-lab-mgmt/issues/506 
![data_layer_error](https://github.com/user-attachments/assets/6029ecb5-cf75-4587-a2e3-e351c5375170)

I agree with you about adding a conditional check before pushing, and will modify the code accordingly. Thanks.

Reply:
Raj, I tried to re-create the error by removing the global setting commit, but somehow I am not able to reproduce the bug. Now, if I make changes with just conditional checks instead of global settings, I will not be able to test it. Kindly advice if I should go ahead with these changes, then I will commit accordingly.